contrib/checkbox-ce-oem/checkbox-provider-ce-oem/bin/gst_resources_generator.py

```diff
@@ -0,0 +1,176 @@
+        width: dict,
+        height: dict,
+        color_space: str,
+        source_format: str,
+    ) -> dict:
+        """
+        Generate a resource item dictionary for
+        gst_v4l2_video_decoder_md5_checksum_comparison scenario
+        """
+        name = "{}x{}-{}-{}".format(width, height, decoder_plugin, color_space)
+        golden_sample_file = "{}/video_golden_samples/{}.{}".format(
+            self._args.gst_testing_data_path, name, source_format
+        )
+        golden_md5_checkum_file = "{}/{}/golden_md5_checksum/{}/{}.md5".format(
+            self._args.gst_testing_data_path,
+            self._current_scenario_name,
+            self._args.gst_conf_name,
+            name,
+        )
+
+        returned_dict = {
+            "scenario": self._current_scenario_name,
+            "decoder_plugin": decoder_plugin,
+            "width": width,
+            "height": height,
+            "color_space": color_space,
+            "golden_sample_file": golden_sample_file,
+            "golden_md5_checkum_file": golden_md5_checkum_file,
+        }
+
+        return returned_dict
+
+    def gst_v4l2_video_decoder_md5_checksum_comparison(
+        self, scenario_data: List[Dict]
+    ) -> None:
+        for item in scenario_data:
+            self._resource_items.extend(
+                [
+                    self._v4l2_video_decoder_md5_checksum_comparison_helper(
+                        decoder_plugin=item["decoder_plugin"],
+                        width=resolution["width"],
+                        height=resolution["height"],
+                        color_space=color_space,
+                        source_format=item["source_format"],
+                    )
+                    for resolution in item["resolutions"]
+                    for color_space in item["color_spaces"]
+                ]
+            )
+
+    def main(self):
+        for scenario in self._scenarios:
+            self._current_scenario_name = scenario
+            getattr(self, scenario)(self._scenarios[scenario])
+            self._dump_resources()
+
+    def _dump_resources(self):
+        """
+        Prints each key-value pair from the self._resource_items in the format
+        "key": "value".
+        """
+        for item in self._resource_items:
+            for key, value in item.items():
+                print("{}: {}".format(key, value))
+            print()
+        print()
+        # Renew for next scenario
+        self._resource_items = []
+
+
+def main() -> None:
+    try:
+        data_dir = os.environ["PLAINBOX_PROVIDER_DATA"]
+    except KeyError:
+        raise SystemExit("PLAINBOX_PROVIDER_DATA variable not set")
+    args = register_arguments()
+
+    # All gstreamer related configs should be put into gstreamer-test-confs
+    # direcotry by design
+    conf_path = os.path.join(
+        data_dir, "gstreamer-test-confs", "{}.json".format(args.gst_conf_name)
+    )
+    scenarios = load_json_config(file_path=conf_path)
```

Comment:
I would suggest we move this part in the GstResources.__init__ function.
```
    # All gstreamer related configs should be put into gstreamer-test-confs
    # direcotry by design
    conf_path = os.path.join(
        data_dir, "gstreamer-test-confs", "{}.json".format(args.gst_conf_name)
    )
    scenarios = load_json_config(file_path=conf_path)
```

Reply:
Update, please check it